File: src/6.ts
// Computes the difference between the sum of the squares of the first n natural numbers and the square of the sum of
// the first n natural numbers.
//
// See {@link https://projecteuler.net/problem=6}.
import _ from 'lodash';

export default function compute(limit: number) {
  const a = _.range(limit + 1)
    .map(n => n ** 2)
    .reduce((a, b) => a + b, 0);
  const b = _.range(limit + 1).reduce((a, b) => a + b, 0) ** 2;
  return b - a;
}

File: project_euler/6.py
# Problem 6: Sum Square Difference
#
# Find the difference between the sum of the square of the first one hundred natural numbers and the square of the sum.
#
# See https://projecteuler.net/problem=6
from typing import cast
from toolz import curried, pipe


def run() -> int:
    a = pipe(
        range(101),
        curried.map(lambda n: n**2),
        curried.reduce(lambda a, b: a + b),
    )
    b = pipe(
        range(101),
        curried.reduce(lambda a, b: a + b),
        lambda n: n**2,
    )

    a = cast(int, a)
    b = cast(int, b)
    return b - a


def test_run():
    assert run() == 25164150
